test/IDE/complete_precedencegroup.swift

```diff
@@ -0,0 +1,117 @@
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=ASSOCIATIVITY_1 | %FileCheck %s -check-prefix=ASSOCIATIVITY
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=ASSOCIATIVITY_2 | %FileCheck %s -check-prefix=ASSOCIATIVITY
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=ASSIGNMENT_1 | %FileCheck %s -check-prefix=ASSIGNMENT
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=ASSIGNMENT_2 | %FileCheck %s -check-prefix=ASSIGNMENT
+
+infix operator +++: #^PRECEDENCE_GROUP_1^#
+
+precedencegroup MyPrecedence1 {
+  #^ATTRIBUTE_LIST_1^#
+}
+precedencegroup MyPrecedence2 {
+  associativity: left #^ATTRIBUTE_LIST_2^#
+}
+precedencegroup MyPrecedence3 {
+  higherThan: AdditionPrecedence
+  lowerThan: MultiplicationPrecedence
+  #^ATTRIBUTE_LIST_3^#
+}
+precedencegroup MyPrecedence4 {
+  assignment: true
+  #^ATTRIBUTE_LIST_4^#
+  higherThan: AdditionPrecedence
+  associativity: right
+  lowerThan: MultiplicationPrecedence
+}
+
+precedencegroup MyPrecedence5 {
+  associativity: #^ASSOCIATIVITY_1^#
+  assignment: false
+}
+precedencegroup MyPrecedence6 {
+  associativity: left
+  lowerThan: #^PRECEDENCE_GROUP_2^#
+  higherThan: AdditionPrecedence
+}
+precedencegroup MyPrecedence7 {
+  associativity: none
+  assignment: false
+  lowerThan: AdditionPrecedence
+  higherThan: #^PRECEDENCE_GROUP_3^#
+}
+precedencegroup MyPrecedence8 {
+  associativity: right
+  assignment: #^ASSIGNMENT_1^#
+}
+
+// Test that we have completions despite successive attributes being invalid
+precedencegroup MyPrecedence9 {
+  associativity: #^ASSOCIATIVITY_2^#
+  assignment: 5
+  higherTh:
+}
+precedencegroup MyPrecedence10 {
+  assignment: #^ASSIGNMENT_2^#
+  lower:
+  associativity: true
+}
+precedencegroup MyPrecedence11 {
+  higherThan: #^PRECEDENCE_GROUP_4^#
+  associativity: 3
+  lowerTh:
+}
+
+// Test that we see new precedence group declarations
+operator <><>: #^PRECEDENCE_GROUP_CURR_FILE^#
+
+
+// ATTRIBUTE_LIST: Begin completions, 4 items
+// ATTRIBUTE_LIST: Keyword/None: associativity; name=associativity
+// ATTRIBUTE_LIST: Keyword/None: higherThan;    name=higherThan
+// ATTRIBUTE_LIST: Keyword/None: lowerThan;     name=lowerThan
+// ATTRIBUTE_LIST: Keyword/None: assignment;    name=assignment
+
+// ASSOCIATIVITY: Begin completions, 3 items
+// ASSOCIATIVITY: Keyword/None: none;   name=none
+// ASSOCIATIVITY: Keyword/None: left;   name=left
+// ASSOCIATIVITY: Keyword/None: right;  name=right
+
+// ASSIGNMENT: Begin completions, 2 items
+// ASSIGNMENT: Keyword[false]/None: false; name=false
+// ASSIGNMENT: Keyword[true]/None: true;   name=true
+
+// PRECEDENCE_GROUP: Begin completions
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: BitwiseShiftPrecedence; name=BitwiseShiftPrecedence
```

Comment:
Instead of checking all the decls from the stdlib, I suggest just checking one or two (e.g. Assignment and Comparison).  We prefer not to rely too much on stdlib details in code-completion tests.